velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -328,6 +328,25 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
           // the "repeated info" which we are ignoring here, hence we set the
           // isRepeated to true eg
           // https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists
+          if (schemaElement.converted_type == thrift::ConvertedType::LIST &&
+              (child->type()->kind() == TypeKind::MAP ||
```

Comment:
could you share more details why we need 
```
|| schemaElement.repetition_type == thrift::FieldRepetitionType::REPEATED
```
? I don't see it in original diff? https://github.com/facebookincubator/velox/pull/9278/files

Reply:
Ahh this extra check is needed to handle nested array cases. So the first check handles array(map) and the second check handles array(array). When I created the previous diff we had not yet fixed array(array) case. Let me add a test case for that as well.